### ISSUE TITLE
[Feature] Can a model with a weight of BF16 be adapted based on an existing framework?

### ISSUE THREAD

**Issue by YoctoHan** [InternLM/lmdeploy#385]
### Motivation

Dear experts, I have been studying and learning your open source project lmdeploy recently. A lot of design and implementation in this project have given me great help and inspiration.

I want to change my model to fastertransformer back-end, but fastertransformer itself does not support GQA, I find that lmdeploy provides this support, so I want to refer to the implementation of lmdeploy, but the weight of my model is BF16. I observe that currently lmdeploy does not support this type. If I want to modify the framework to support the BF16 type, what parts of the code should I focus on and change?

**Comment by lzhangzz** [comment]
`deploy.py` will convert bf16 tensor to half tensor automatically.

**Comment by YoctoHan** [comment]
> `deploy.py` will convert bf16 tensor to half tensor automatically.

Yes, I have also noticed that this script can be used for weight conversion, but I currently only want to migrate the MQA part of the code into my model, which organizes the model weights in other ways so I cannot directly use this script for conversion. I have tried to convert my model weight to half type for inference, but in some cases there will be an upoverflow, resulting in inference interruption, so the best way for me to implement the current BF16 type is still to reason, will this be very complicated?

I will try to deploy my own model with turboMind later, and I have found that many of your clever designs meet my expectations, and will do FP16 reasoning and try to solve the upoverflow problem.
